curriculum/challenges/english/10-coding-interview-prep/data-structures/insert-an-element-into-a-max-heap.md

```diff
@@ -110,8 +110,12 @@ assert(
     test.insert(700);
     test.insert(32);
     test.insert(51);
-    let result = test.print();
-    return result.length == 5 ? result[0] == 700 : result[1] == 700;
+    test.insert(800);
+    let result = JSON.stringify(test.print());
+    let solution = JSON.stringify([null,800,51,700,32,50,100]);
+    let solutionWithoutNull = JSON.stringify([800,51,700,32,50,100]);
+
+    return (result.length == 6) ? (result == solution) : (result == solutionWithoutNull);
```

Comment:
`result` is here stringified version of the heap array, so length `6` will neither be right for version with null nor without.

Assuming length of the heap array will be checked, it should be compared to `7`, as that's how are ordered comparisons with solutions later in the ternary operator.

And from the nitpicking - those variables can be declared with `const`.